File: Java/DecoratorPattern/src/coffee/Coffee.java
package coffee;

/**
 * Author: Utkarsh Dayal @ descifrador17
 * Date: 27-05-2021
 * Time: 04:20 AM
 */
public abstract class Coffee {
    String description = "Coffee";

    public String getDescription(){
        return this.description;
    }

    public abstract double cost();
}

File: Kotlin/DecoratorPattern/src/coffee/Coffee.kt
package coffee

/**
 *
 *   Author: Utkarsh Dayal @ descifrador17
 *   Date: 27-05-2021
 *   Time: 04:55 AM
 *
 */
abstract class Coffee {
    var description: String = "Coffee"

    abstract fun cost(): Double

    open fun getCoffeeDescription(): String{
        return description
    }
}